### ISSUE TITLE
fix(ux): mandatory field on External Reference is not correctly handled on the creation form

### ISSUE THREAD

**Issue by Lhorus6** [LuatixHQ/opencti#8612]
## Description

When I create an External Reference the form doesn't correctly display the information that the URL is mandatory

![image](https://github.com/user-attachments/assets/bd217f25-2c25-46de-ad27-ed734063be38)

## Environment

OCTI 6.3.4

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Go on a Report
2. Scroll down to add an external ref
3. Create an Ext Ref -> fill in the "name" field but not the "URL" and try to create

## Expected Output

Example with the name field:

![image](https://github.com/user-attachments/assets/6df52f80-3090-4340-8013-a06061629c44)

## Additional information

Maybe related, it's not noted as mandatory in the CSV mapper either:

![image](https://github.com/user-attachments/assets/f583d2f7-2753-4b7d-9ed5-02bef96dcf01)

**Comment by romain-filigran** [comment]
Two corrective actions to handle: 

1. Mark the URL field as mandatory in the external reference creation form
2.  Mark the URL field as mandatory in the CSV mapper form

**Comment by Kedae** [comment]
URL is not mandatory, you have to put either an External Id, a URL or a File I think ?

**Comment by romain-filigran** [comment]
You right @Kedae.

**Comment by Lhorus6** [comment]
Oh ok, it's mandatory: `Name AND (url OR external id OR file)` ?

It may be complicated, but it has to be understandable.

**Comment by Kedae** [comment]
May be we can try to address this matter with the "mandatory fields" work from community ? @frapuks @Jipegien @romain-filigran

**Comment by frapuks** [comment]
Currently, we have 3 fields touched by this issue :
- Source name : always mandatory
- External ID
- Url

We need to have an external id OR an url.

The FBI PR is to display a star ater mandatory fields, and take account of the custom mandatory properties. But, we don't have customizable fields on External Ref.

I try to add a condition to set a external id as mandatory if url is empty :
![image](https://github.com/user-attachments/assets/465bb360-6835-4f18-9c86-d7f466aeeb6c)
But we can't use the same solution for the second field url because this is a cycling condition...

=> https://github.com/OpenCTI-Platform/opencti/compare/master...issue/8612

**Comment by frapuks** [comment]
[My FBI PR about 1st step of mandatory fields](https://github.com/OpenCTI-Platform/opencti/pull/6972) is ready to merge. Do i need to ask him to manage this part or i can merge this PR ?

**Comment by SarahBocognano** [comment]
Was it fixed with issue/8612 ?

**Comment by frapuks** [comment]
No, i create this branch only for trying a solution for a fast example. I don't work more on this issue.
